common/codex32.c

```diff
@@ -0,0 +1,338 @@
+#include "config.h"
+#include <assert.h>
+#include <bitcoin/chainparams.h>
+#include <ccan/tal/str/str.h>
+#include <common/bech32_util.h>
+#include <common/bolt12.h>
+#include <common/codex32.h>
+#include <common/bolt12_merkle.h>
+#include <common/configdir.h>
+#include <common/features.h>
+#include <math.h>
+#include <secp256k1_schnorrsig.h>
+#include <time.h>
+
+const uint8_t LOG[32] = {
+     0,  0,  1, 14,  2, 28, 15, 22,
+     3,  5, 29, 26, 16,  7, 23, 11, 
+     4, 25,  6, 10, 30, 13, 27, 21,
+    17, 18,  8, 19, 24,  9, 12, 20,
+};
+
+const uint8_t LOG_INV[31] = {
+     1,  2,  4,  8, 16,  9, 18, 13,
+    26, 29, 19, 15, 30, 21,  3,  6,
+    12, 24, 25, 27, 31, 23,  7, 14,
+    28, 17, 11, 22,  5, 10, 20,
+};
+
+static void addition_gf32(uint8_t *x, uint8_t y) {
+	*x = *x ^ y;
+	return;
+}
```

Comment:
```suggestion
static void addition_gf32(uint8_t *x, uint8_t y) 
{
	*x = *x ^ y;
	return;
}
```

All the function need this change :)

Reply:
Done, Thanks!